marimo/_ast/names.py

```diff
@@ -1,5 +1,13 @@
+import hashlib
+
 DEFAULT_CELL_NAME = "__"
 
 
 def is_default_cell_name(name: str) -> bool:
-    return name.startswith("__") and (name == "__" or name[2:].isdigit())
+    return name.startswith("__") and (name == "__" or len(name) == 6)
+
+
+def default_cell_name_from_code(code: str) -> str:
+    return (
+        f"{DEFAULT_CELL_NAME}{hashlib.sha256(code.encode()).hexdigest()[:4]}"
+    )
```

Comment:
I think @dmadisetti's proposal was to use the hashing mechanism developed for caching, but not sure -- will let him comment.

Reply:
yea that would have the benefit of knowing which cells could change transitively, but that also increases the noise per commit (e.g. adding a new import could change each function name)

Reply:
it could also be configurable (in future) or part of strict-mode